pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,21 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// BIOSConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type BIOSConfig struct {
+
+	// Select the boot mode of the system. Allowed values are Uefi, LegacyBios
+	// +kubebuilder:validation:Enum=Uefi,LegacyBios
+	BootMode string `json:"bootMode,omitempty"`
+
+	// Hyper-threading technology allows a single physical processor core to appear as several
+	// logical processors. This supports following options: true, false.
+	HyperthreadingEnabled bool `json:"hyperthreadingEnabled,omitempty"`
+
+	// Supports the virtualization of platform hardware. This supports following options: true, false.
+	VirtualizationEnabled bool `json:"virtualizationEnabled,omitempty"`
```

Comment:
We should think about the naming for boolean options like this one and the previous one so we can establish a good naming convention.

FooEnabled can be read as either "foo is enabled" or "foo should be enabled".

EnableFoo or UseFoo are more imperative forms. I'm not sure about using those in a declarative interface.

For the Online flag, I didn't include a verb at all, because I thought the verb was implicit in a declarative interface ("make the online setting match what I give you in this field"). Does that approach make sense, too? Should we call this flag "Virtualization" and the previous one "Hyperthreading"?

I'm curious about the opinions of other reviewers.